src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputNoWhitespaceBeforeAnnotationsAndEllipsis.java

```diff
@@ -0,0 +1,15 @@
+package com.google.checkstyle.test.chapter4formatting.rule462horizontalwhitespace;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE_USE)
+@interface NonNull {
+
+}
+
+public class InputNoWhitespaceBeforeAnnotationsAndEllipsis {
+    void test1(String @NonNull ... param) {} // warn
+
+    void test2(String @NonNull... param) {}
```

Comment:
In issue I provided several usecases, please use them all and be creative and do even more, the weirder code the better.

Reply:
done

Reply:
please make Input like https://github.com/checkstyle/checkstyle/issues/6707#issuecomment-620635286
We need more test cases.

Reply:
done

Reply:
done